.github/workflows/e2e_test_ci.yml

```diff
@@ -0,0 +1,335 @@
+
+      - name: Wait for Spice runtime healthy
+        working-directory: test_app
+        timeout-minutes: 1
+        run: |
+          while [[ "$(curl -s http://localhost:3000/health)" != "ok" ]]; do sleep 1; done
+
+      - name: Add spiceai/quickstart
+        working-directory: test_app
+        run: |
+          spice add spiceai/quickstart
+          cat spicepod.yaml
+
+      - name: Check datasets
+        working-directory: test_app
+        run: |
+          response=$(curl http://localhost:3000/v1/datasets)
+          length=$(echo $response | jq 'if type=="array" then length else empty end')
+          if [[ $length -ne 1 ]]; then
+            echo "Unexpected response: $response, expected 1 dataset"
+            exit 1
+          fi
+
+      - name: Check tables
+        working-directory: test_app
+        run: |
+          response=$(curl -X POST \
+            -H "Content-Type: text/plain" \
+            -d "show tables;" \
+            http://localhost:3000/v1/sql
+          )
+          tables=$(echo $response | jq 'if type=="array" then length else empty end')
+          if [[ $length -ne 5 ]]; then
+            echo "Unexpected response: $response, expected 5 tables"
+            exit 1
+          fi
+
+  test_quickstart_spiceai:
+    name: "E2E Test: SpiceAI quickstart, using ${{ matrix.target_os }}-${{ matrix.target_arch }}"
+    runs-on: ${{ matrix.runner }}
+    needs: build
+
+    strategy:
+      matrix:
+        include:
+          - name: Linux x64
+            runner: ubuntu-latest
+            target_os: linux
+            target_arch: x86_64
+            target_arch_go: amd64
+          # - name: macOS aarch64 (Apple Silicon)
+          #   runner: macos-14
+          #   target_os: darwin
+          #   target_arch: aarch64
+          #   target_arch_go: arm64
+          # - name: macOS x64 (Intel)
+          #   runner: rust-osx-x64
+          #   target_os: darwin
+          #   target_arch: x86_64
+          #   target_arch_go: amd64
+          # - name: Windows x64
+          #   runner: rust-windows-x64
+          #   target_os: windows
+          #   target_arch: x86_64
+          #   target_arch_go: amd64
+
+    steps:
+      - name: download artifacts - build_${{ matrix.target_os }}_${{ matrix.target_arch }}
+        uses: actions/download-artifact@v4
+        with:
+          name: build_${{ matrix.target_os }}_${{ matrix.target_arch }}
+          path: ./build
+
+      - name: Install spice
+        run: |
+          chmod +x ./build/spice
+          chmod +x ./build/spiced
+          ls -la $HOME
+          ls -la $HOME/build
+          echo "$HOME/build" >> $GITHUB_PATH
+
+      - name: Check spice version
+        run: spice version
+
+      - name: Init spice app
+        run: |
+          spice init test_app
+
+      - name: Configure spice app
+        working-directory: test_app
+        run: |
+          # equal to "spice dataset configure"
+          mkdir -p datasets/eth_recent_blocks
```

Comment:
Let's simulate the same user command via pipe

echo -e "eth_recent_blocks\neth recent logs\nspice.ai/eth.recent_blocks\ny" | spice dataset configure

Reply:
@sgrebnov good point, thanks. Let's add that in separate PR, merging this as is now.